src/groupByKey.ts

```diff
@@ -2,6 +2,19 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const groupedByKey: GroupsMap<T> = {};
+  const keysInGrouped: any[] = [];
```

Comment:
using 'any' type might not be the best approach.

Reply:
Yep, it's a bad idea. Don't use `any`